### ISSUE TITLE
Spaces are encoded differently with the polyfill

### ISSUE THREAD

**Issue by csvn** [lifaon74/url-polyfill#20]
When using a form submit to search, forms encode spaces as `+`:

```html
<form action="/search">
  <input type="hidden" name="q" value="hello world">
</form>
<!-- Submit results in URL: "<origin>/search?q=hello+world" -->
```

When decoding this URL and fetching the `q` parameter the result differs between browsers with and without polyfill:

```javascript
// href => <origin>/search?q=hello+world
const url = new URL(location.href);
console.log(url.searchParams.get('q'));
// Prints "hello world" in latest Chrome
// Prints "hello+world" in Edge with this polyfill
```

I looked up how parameters with `URLSearchParams` should be handled, and it seems the search params should use form encoding:

1. Set should update entry list, and then do the update steps [(whatwg)](https://url.spec.whatwg.org/#dom-urlsearchparams-set)
2. The update steps should serialize all search params [(whatwg)](https://url.spec.whatwg.org/#concept-urlsearchparams-update)
3. Serialization uses the form encode serializer [(whatwg)](https://url.spec.whatwg.org/#concept-urlencoded-serializer)
4. The serialization above uses a byte serializer where spaces should be replaced by `+` for both search param name and values [(whatwg)](https://url.spec.whatwg.org/#concept-urlencoded-byte-serializer)

In short, the [steps for encoding that are listed on whatwg](https://url.spec.whatwg.org/#concept-urlencoded-byte-serializer) seems to follow `encodeURIComponent()` except for `+` which is encoded as `%20`. I suggest that a method that respects space as `+` to be followed for decoding and encoding search params and their values.

**Comment by csvn** [comment]
I was thinking about doing a PR for this. Only the below two changes would need to be handled for it to be more inline with the spec and modern browsers.

* https://github.com/lifaon74/url-polyfill/blob/master/url.ts#L142
  Replace `encodeURIComponent(x)` with `encodeURIComponent(x.replace(/ /g, '+'))`
* https://github.com/lifaon74/url-polyfill/blob/master/url.ts#L65-L66
  Replace `decodeURIComponent(x)` with `decodeURIComponent(x).replace(/\+/g, ' ')`

But I am unsure of where to put my code. In `url.ts`? In `url.js`? Both? Doesn't look like `url.ts` is used in a build process, and some changes seems to be done in both files. I cannot see Typescript as a dependency either (for compiling `ts` to `js` with `tsc`).

**Comment by lifaon74** [comment]
As specified everywhere (but it seems not enough because I have/had a lot of request about url.ts). **THIS LIB IS NOT INTENDED TO BE USED IN YOUR PROJETS**, its only an experimental file. That's said, could you tell me if the bug come from url-polyfill (the correct file to use) ? If yes, i'll be glad to add your changes ;)

**Comment by csvn** [comment]
@lifaon74 We are just importing `url-polyfill` via `import 'url-polyfill';', so the `main` file `url-polyfill.js` should be the one we are using currently.

The questions above was because I did not know if the other files were generated from the others 😊
